frontend/javascripts/types/api_flow_types.ts

```diff
@@ -199,12 +199,26 @@ export type APIDatasetCompactWithoutStatus = Pick<
   | "tags"
   | "isUnreported"
 >;
-export type APIDatasetCompact = APIDatasetCompactWithoutStatus & {
+export type APIDatasetCompact = APIDatasetCompactWithoutStatusAndLayerNames & {
   id?: string;
   status: MutableAPIDataSourceBase["status"];
+  colorLayerNames: Array<string>;
+  segmentationLayerNames: Array<string>;
 };
 
 export function convertDatasetToCompact(dataset: APIDataset): APIDatasetCompact {
+  let colorLayerNames = [];
+  let segmentationLayerNames = [];
+  if (dataset.dataSource) {
+    for (const layer of dataset.dataSource.dataLayers) {
+      if (layer.category === "segmentation") {
+        segmentationLayerNames.push(layer.name);
+      } else if (layer.category === "color") {
+        colorLayerNames.push(layer.name);
+      }
+    }
+  }
```

Comment:
```suggestion
  const [colorLayerNames, segmentationLayerNames] = _.partition(
    dataset.dataSource.dataLayers,
    (layer) => layer.category === "segmentation",
  ).map((layers) => layers.map((layer) => layer.name).sort());
```

not sure why you checked `dataset.dataSource`? For an `APIDataset`, the datasource should always exist (unlike `APIMaybeUnimportedDataset`).